lib/dom-parser.js

```diff
@@ -49,98 +43,82 @@ function normalizeLineEndings(input) {
 
 /**
  * @typedef DOMParserOptions
- * @property {typeof conventions.assign} [assign=Object.assign || conventions.assign]
- * The method to use instead of `Object.assign` (or if not available `conventions.assign`),
- * which is used to copy values from the options before they are used for parsing.
+ * @property {typeof assign} assign
+ * The method to use instead of `Object.assign`
+ * (or if not available `conventions.assign`), which is used to copy values from the options before they are used for parsing.
+ * Default is `Object.assign || conventions.assign`
```

Comment:
It's fine to move the description of the default value to the text, but the parameter needs to still be marked as optional.

And completely unrelated to this PR I just realized that the described default is not up to date with the code, so it should be changed to
```suggestion
 * @property {typeof assign} [assign]
 * The method to use instead of `conventions.assign`, which is used to copy values from `options` before they are used for parsing.
 * Default is `conventions.assign`
```